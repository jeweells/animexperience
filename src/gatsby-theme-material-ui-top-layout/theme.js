import { createTheme } from '@material-ui/core/styles'

export const mainColor = 'rgb(105, 190, 213)'

const theme = createTheme({
    typography: {
        fontFamily: ['Quicksand', 'sans-serif'].join(','),
    },
    palette: {
        primary: {
            main: '#292d33',
            contrastText: '#e9ebf0',
        },
        secondary: {
            main: '#3c3f43',
        },
        text: {
            primary: '#e9ebf0',
            secondary: '#e9ebf0',
            disabled: '#ffffff82',
        },
    },
    overrides: {
        MuiButtonBase: {
            root: {
                borderRadius: undefined,
            },
        },
        MuiButton: {
            root: {
                color: '#e9ebf0',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                height: 36,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },

            text: {
                color: '#e9ebf0',
                textTransform: 'none',
                padding: '8px 12px',
            },
        },
        MuiIconButton: {
            root: {
                flexShrink: 0,
                height: 36,
                minWidth: 36,
                color: '#e9ebf0',
                padding: '8px 12px',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },
        },
        MuiCssBaseline: {
            '@global': {
                html: {
                    WebkitFontSmoothing: 'auto',
                },
                body: {
                    color: undefined,
                    margin: undefined,
                    fontSize: undefined,
                    fontFamily: undefined,
                    fontWeight: undefined,
                    backgroundColor: undefined,
                    lineHeight: undefined,
                },
            },
        },
        MuiSkeleton: {
            root: {
                backgroundColor: 'rgba(255, 255, 255, 0.11)',
            },
            pulse: {
                animation: 'MuiSkeleton-keyframes-pulse 3s ease-in 0.5s infinite',
            },
        },
    },
})

export default theme
