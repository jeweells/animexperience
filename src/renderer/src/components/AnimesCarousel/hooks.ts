import { useMemo, useState, HTMLAttributes } from 'react'
import useResizeObserver from 'use-resize-observer'

export const useSliding = (gap: number, countElements: number, prevButtonWidth: number) => {
  const { ref: containerRef, width: containerWidth = 1 } = useResizeObserver<HTMLDivElement>()
  const { ref: scrollerRef, width: scrollerWidth = 1 } = useResizeObserver<HTMLDivElement>()
  const [viewed, setViewed] = useState(0)
  const availableScrollerWidth = scrollerWidth - prevButtonWidth

  const elementWidth = useMemo(() => {
    return (containerWidth - gap * (countElements - 1)) / countElements
  }, [countElements, gap, containerWidth])
  const totalInViewport = useMemo(() => {
    const elementWidthWithGap = elementWidth + gap

    let visibleElementsCount = Math.floor(availableScrollerWidth / elementWidthWithGap)
    if (elementWidthWithGap * visibleElementsCount + elementWidth <= availableScrollerWidth) {
      visibleElementsCount++
    }
    return visibleElementsCount
  }, [availableScrollerWidth, elementWidth])

  const [sliding, setSliding] = useState<boolean>(false)
  const handlePrev = () => {
    if (!sliding) {
      setViewed((viewed) => Math.min(countElements, Math.max(viewed - totalInViewport, 0)))
      setSliding(true)
    }
  }

  const handleNext = () => {
    if (!sliding) {
      setViewed((viewed) => Math.min(countElements, Math.max(viewed + totalInViewport, 0)))
      setSliding(true)
    }
  }

  const distance = useMemo(() => {
    const elementWithGap = elementWidth + gap
    return -prevButtonWidth + elementWithGap * viewed
  }, [viewed, elementWidth, gap, prevButtonWidth])

  const slideProps = useMemo<Partial<HTMLAttributes<HTMLElement>>>(
    () => ({
      style: {
        transform: `translate3d(${-distance}px, 0, 0)`,
        // These will be unset when the transition ends
        pointerEvents: sliding ? 'none' : 'all'
      }
    }),
    // Do not depend it on distance; very important! Otherwise, 1 frame will be inconsistent
    [sliding]
  )

  const hasPrev = distance > 0
  const hasNext = viewed + totalInViewport < countElements

  return {
    scrollerRef,
    handlePrev,
    handleNext,
    slideProps,
    containerRef,
    hasPrev,
    hasNext,
    sliding,
    viewed,
    totalInViewport,
    onSlidingStart: () => {
      setSliding(true)
    },
    onSlidingComplete: () => {
      setSliding(false)
    }
  }
}

const defaultSizes = {
  gap: 16,
  containerWidth: '100vw',
  navigationWidth: 72
}

export const useSizes = () => {
  return defaultSizes
}
