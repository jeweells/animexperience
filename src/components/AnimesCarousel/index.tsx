import React, { Fragment } from 'react'
import { Transition } from 'react-transition-group'
import styled from 'styled-components'
import { range } from '../../utils'
import { NavigationButton } from './components/NavigationButton'
import { useSizes, useSliding } from './hooks'

import { CarouselTitleWithLoading } from '~/src/components'

const Scroller = styled.div`
    overflow-x: hidden;
    position: relative;
`

const Items = styled.div<{ gap: number }>`
    display: flex;
    width: min-content;
    gap: ${(props) => props.gap}px;
`

export type AnimesCarouselProps = {
    render(info: { index: number; visible: boolean; sliding: boolean }): React.ReactNode
    loading?: boolean
    count: number
    title?: string
}

export const AnimesCarousel: React.VFC<AnimesCarouselProps> = React.memo(
    ({ count, loading, title, render }) => {
        const { gap, navigationWidth, containerWidth } = useSizes()
        const {
            containerRef,
            scrollerRef,
            handleNext,
            handlePrev,
            hasNext,
            hasPrev,
            slideProps,
            sliding,
            viewed,
            totalInViewport,
            onSlidingComplete,
        } = useSliding(gap, count, navigationWidth)
        const duration = 400
        if (!loading && count === 0) return null
        return (
            <Fragment>
                <CarouselTitleWithLoading title={title} loading={loading} />
                <Scroller ref={scrollerRef} style={{ width: containerWidth }}>
                    <NavigationButton
                        onClick={handlePrev}
                        disabled={loading || !hasPrev}
                        direction={'left'}
                    />
                    <Transition
                        in={sliding}
                        unmountOnExit={false}
                        onEntered={onSlidingComplete}
                        timeout={400}
                    >
                        {() => (
                            <Items
                                ref={containerRef}
                                gap={gap}
                                style={{
                                    transition: `all ${duration}ms`,
                                    ...slideProps.style,
                                }}
                            >
                                {range(count).map((x) => {
                                    return render({
                                        index: x,
                                        visible:
                                            viewed <= x && x < viewed + totalInViewport,
                                        // This way a few range of elements is going to re-render
                                        sliding:
                                            sliding &&
                                            viewed - totalInViewport <= x &&
                                            x < viewed + 2 * totalInViewport,
                                    })
                                })}
                            </Items>
                        )}
                    </Transition>
                    <NavigationButton
                        onClick={handleNext}
                        disabled={loading || !hasNext}
                        direction={'right'}
                    />
                </Scroller>
            </Fragment>
        )
    },
)

AnimesCarousel.displayName = 'AnimesCarousel'

export default AnimesCarousel
