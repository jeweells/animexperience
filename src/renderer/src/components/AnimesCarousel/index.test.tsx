import AnimesCarousel, { AnimesCarouselProps } from './index'
import useResizeObserver from 'use-resize-observer'
import { ThemeProvider } from '@mui/material'
import theme from '../../theme'

describe('AnimesCarousel', () => {
  let props: AnimesCarouselProps
  const useResizeObserverMock = useResizeObserver as jest.Mock
  const invokeMock = window.electron.ipcRenderer.invoke as jest.Mock
  const imageUrl = 'fake_image_url'

  const getComponent = (override: Partial<AnimesCarouselProps> = {}) => {
    return (
      <ThemeProvider theme={theme}>
        <AnimesCarousel {...props} {...override} />
      </ThemeProvider>
    )
  }

  beforeEach(() => {
    props = {
      count: 5,
      loading: false,
      title: 'Sample title',
      render: (info) => <div key={info.index}>{JSON.stringify(info, null, 2)}</div>
    }

    useResizeObserverMock.mockReturnValue({ ref: jest.fn(), width: 100, height: 100 })
    invokeMock.mockResolvedValue(imageUrl)
  })

  it.each([[false], [true]])('renders default when %s', (loading) => {
    const wrapper = render(getComponent({ loading }))
    expect(wrapper.baseElement).toMatchSnapshot()
  })
})
