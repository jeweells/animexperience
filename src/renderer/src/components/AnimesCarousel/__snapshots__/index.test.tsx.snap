// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimesCarousel renders default when false 1`] = `
.emotion-0 {
  font-weight: 500;
  color: #e7e7e7;
  font-size: 1.2rem;
  line-height: 1;
  text-shadow: 0 0 2px black,0 0 6px black;
  margin-bottom: 24px;
}

.emotion-1 {
  overflow-x: hidden;
  position: relative;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 10;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0!important;
  background-image: linear-gradient(
      to left,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-3:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-3:hover {
  opacity: 0!important;
  background-image: linear-gradient(
        to left,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  gap: 16px;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0.7!important;
  background-image: linear-gradient(
      to right,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-7:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-7:hover {
  opacity: 1!important;
  background-image: linear-gradient(
        to right,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

.emotion-9 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <h2
      class="emotion-0"
      style="margin-left: 72px; margin-bottom: 24px;"
    >
      Sample title
    </h2>
    <div
      class="emotion-1"
      style="width: 100vw;"
    >
      <div
        class="emotion-2"
        style="left: 0px; width: 72px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-3"
          data-testid="navigation-button-button"
          direction="left"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-4"
            data-testid="KeyboardArrowLeftRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.71 15.88 10.83 12l3.88-3.88c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0L8.71 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .38-.39.39-1.03 0-1.42"
            />
          </svg>
        </button>
      </div>
      <div
        class="emotion-5"
        style="transition: all 400ms; transform: translate3d(72px, 0, 0); pointer-events: all;"
      >
        <div>
          {
  "index": 0,
  "visible": true,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 1,
  "visible": false,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 2,
  "visible": false,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 3,
  "visible": false,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 4,
  "visible": false,
  "sliding": false
}
        </div>
      </div>
      <div
        class="emotion-2"
        style="right: 0px; width: 72px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-7"
          data-testid="navigation-button-button"
          direction="right"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-4"
            data-testid="KeyboardArrowRightRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9.29 15.88 13.17 12 9.29 8.12a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42"
            />
          </svg>
          <span
            class="MuiTouchRipple-root emotion-9"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`AnimesCarousel renders default when true 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translateX(5%);
    -moz-transform: translateX(5%);
    -ms-transform: translateX(5%);
    transform: translateX(5%);
  }

  to {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    opacity: 1;
  }
}

.emotion-0 {
  font-weight: 500;
  color: #e7e7e7;
  font-size: 1.2rem;
  line-height: 1;
  text-shadow: 0 0 2px black,0 0 6px black;
  margin-bottom: 24px;
}

.emotion-1 {
  -webkit-animation: animation-0 1500ms;
  animation: animation-0 1500ms;
}

.emotion-2 {
  overflow-x: hidden;
  position: relative;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 10;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0!important;
  background-image: linear-gradient(
      to left,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-4:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-4:hover {
  opacity: 0!important;
  background-image: linear-gradient(
        to left,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

.emotion-5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  gap: 16px;
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
  height: 100%;
  border-radius: 0;
  background-color: transparent!important;
  opacity: 0!important;
  background-image: linear-gradient(
      to right,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.3) 50%,
      rgba(0, 0, 0, 0.4)
    );
  font-size: 3.5rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  padding: 0 24px;
}

.emotion-8::-moz-focus-inner {
  border-style: none;
}

.emotion-8.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-8 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-8:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-8:hover {
    background-color: transparent;
  }
}

.emotion-8.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-8:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-8:hover {
  opacity: 0!important;
  background-image: linear-gradient(
        to right,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.4) 50%,
        rgba(0, 0, 0, 0.8)
      );
}

<body>
  <div>
    <h2
      class="emotion-0"
      style="margin-left: 72px; margin-bottom: 24px;"
    >
      <div
        class="emotion-1"
      >
        <div
          animation="pulse"
          variant="text"
        >
          Skeleton:Mock
        </div>
      </div>
    </h2>
    <div
      class="emotion-2"
      style="width: 100vw;"
    >
      <div
        class="emotion-3"
        style="left: 0px; width: 72px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-4"
          data-testid="navigation-button-button"
          direction="left"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-5"
            data-testid="KeyboardArrowLeftRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.71 15.88 10.83 12l3.88-3.88c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0L8.71 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .38-.39.39-1.03 0-1.42"
            />
          </svg>
        </button>
      </div>
      <div
        class="emotion-6"
        style="transition: all 400ms; transform: translate3d(72px, 0, 0); pointer-events: all;"
      >
        <div>
          {
  "index": 0,
  "visible": true,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 1,
  "visible": false,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 2,
  "visible": false,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 3,
  "visible": false,
  "sliding": false
}
        </div>
        <div>
          {
  "index": 4,
  "visible": false,
  "sliding": false
}
        </div>
      </div>
      <div
        class="emotion-3"
        style="right: 0px; width: 72px;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-8"
          data-testid="navigation-button-button"
          direction="right"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-5"
            data-testid="KeyboardArrowRightRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9.29 15.88 13.17 12 9.29 8.12a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
